internal/webhook/external/refresh.go

```diff
@@ -103,3 +125,61 @@ func refreshWarehouses(
 		},
 	)
 }
+
+func needsRefresh(
+	_ context.Context,
+	subs []kargoapi.RepoSubscription,
+	qualifier string,
+	repoURLs ...string,
+) (*bool, error) {
+	var shouldRefresh bool
+	subs = filterSubsByRepoURL(subs, repoURLs...) // only interested in subs that contain any of the repo URLs.
+	for _, s := range subs {
+		switch {
+		case s.Git != nil:
+			selector, err := commit.NewSelector(*s.Git, nil)
+			if err != nil {
+				return nil, fmt.Errorf("error creating commit selector for Git subscription %q: %w",
+					s.Git.RepoURL, err,
+				)
+			}
+			shouldRefresh = selector.MatchesRef(qualifier)
+		case s.Image != nil:
+			selector, err := image.NewSelector(*s.Image, nil)
+			if err != nil {
+				return nil, fmt.Errorf("error creating image selector for Image subscription %q: %w",
+					s.Image.RepoURL, err,
+				)
+			}
+			shouldRefresh = selector.MatchesTag(qualifier)
+		case s.Chart != nil:
+			selector, err := chart.NewSelector(*s.Chart, nil)
+			if err != nil {
+				return nil, fmt.Errorf("error creating chart selector for Chart subscription %q: %w",
+					s.Chart.RepoURL, err,
+				)
+			}
+			shouldRefresh = selector.MatchesVersion(qualifier)
+		}
+		if shouldRefresh {
+			// exit early if we already found a match
+			return &shouldRefresh, nil
+		}
+	}
+	return &shouldRefresh, nil
```

Comment:
Related to https://github.com/akuity/kargo/pull/4560/files#r2229848911

```suggestion
	}
	return false, nil
```

Reply:
It's gotta be a bool ptr so can't return `false` here and I don't think we should nix the post-switch early return for reasons stated in https://github.com/akuity/kargo/pull/4560#discussion_r2230536902

Reply:
The thread in https://github.com/akuity/kargo/pull/4560#discussion_r2230536902 aside...

Why is this function returning a bool pointer to begin with? It's a bit unusual. Why not a bool?